dialogue-apache-hc5-client/src/main/java/com/palantir/dialogue/hc5/InstrumentedSslConnectionSocketFactory.java

```diff
@@ -0,0 +1,206 @@
+    private static final SafeLogger log = SafeLoggerFactory.get(InstrumentedSslConnectionSocketFactory.class);
+
+    private final Supplier<Socket> rawSocketCreator;
+    private final String[] supportedProtocols;
+    private final String[] supportedCipherSuites;
+    private final String clientName;
+
+    private final Timer connectTimerSuccess;
+    private final Timer connectTimerFailure;
+
+    InstrumentedSslConnectionSocketFactory(
+            String clientName,
+            DialogueClientMetrics dialogueClientMetrics,
+            SSLSocketFactory socketFactory,
+            String[] supportedProtocols,
+            String[] supportedCipherSuites,
+            HostnameVerifier hostnameVerifier,
+            Supplier<Socket> rawSocketCreator) {
+        super(socketFactory, supportedProtocols, supportedCipherSuites, hostnameVerifier);
+        this.supportedProtocols = supportedProtocols;
+        this.supportedCipherSuites = supportedCipherSuites;
+        this.clientName = clientName;
+        this.rawSocketCreator = rawSocketCreator;
+        this.connectTimerSuccess = dialogueClientMetrics
+                .connectionConnect()
+                .clientName(clientName)
+                .result(ConnectionConnect_Result.SUCCESS)
+                .build();
+        this.connectTimerFailure = dialogueClientMetrics
+                .connectionConnect()
+                .clientName(clientName)
+                .result(ConnectionConnect_Result.FAILURE)
+                .build();
+    }
+
+    @Override
+    public Socket createSocket(HttpContext _context) {
+        return rawSocketCreator.get();
+    }
+
+    @Override
+    public Socket connectSocket(
+            Socket socket,
+            HttpHost host,
+            InetSocketAddress remoteAddress,
+            InetSocketAddress localAddress,
+            Timeout connectTimeout,
+            Object attachment,
+            HttpContext context)
+            throws IOException {
+        Preconditions.checkNotNull(host, "host is required");
+        Preconditions.checkNotNull(remoteAddress, "remoteAddress is required");
+        Socket sock = socket != null ? socket : createSocket(context);
+        if (localAddress != null) {
+            sock.bind(localAddress);
+        }
+        try {
+            if (log.isDebugEnabled()) {
+                log.debug(
+                        "Connecting socket to {} ({}) with timeout {}",
+                        SafeArg.of("clientName", clientName),
+                        UnsafeArg.of("remoteAddress", remoteAddress),
+                        SafeArg.of("connectTimeout", connectTimeout));
+            }
+            // Run this under a doPrivileged to support lib users that run under a SecurityManager this allows granting
+            // connect permissions only to this library
+            try {
+                AccessController.doPrivileged((PrivilegedExceptionAction<Object>) () -> {
```

Comment:
nit: since we don't use the returned value, what are your thoughts on

```suggestion
                AccessController.doPrivileged((PrivilegedExceptionAction<Void>) () -> {
```

Reply:
:+1: